extension/extension.proto

```diff
@@ -0,0 +1,177 @@
+syntax = "proto3";
+
+package gnoi.extension;
+
+import "types/types.proto";
+
+option go_package = "github.com/openconfig/gnoi/extension";
+
+option (types.gnoi_version) = "0.1.0";
+
+// Extension is a service for managing extensions on the system.
+// An extension is a collection of files (e.g. RPMs, squashfs)
+// which are installed to modify the behaviour of the services on the
+// system.
+// Only one RPC (SetSignatureVerification, FinalizeExtensions,
+// InstallExtension, UninstallExtension)
+// may be in progress at any given time
+service Extension{
+
+  // SetSignatureVerification sets the methods for which the
+  // system should verify incoming extensions
+  rpc SetSignatureVerification(SigVerificationRequest)
```

Comment:
What's the motivation for this being a separate RPC which then implies there's some state held across requests? It seems like it could just be a flag to `Install...`.

Reply:
This was made a separate RPC as there is state being held across requests :)
the intention is that the signature verification method is global config (e.g. it may mark a single SSL profile as being the chosen profile to verify against). 
It can be set once before any extensions are installed, so that the same config is applied to all extensions installed afterwards (in the case of multiple `InstallExtension` RPCs), rather than relying on the client to re-specify the same option in their requests.